crates/ruma-state-res/src/event_auth.rs

```diff
@@ -557,6 +559,35 @@ fn valid_membership_change(
                 allow
             }
         }
+        // TODO: Does this need to be gated
+        #[cfg(feature = "unstable-pre-spec")]
+        MembershipState::Knock if room_version.allow_knocking => {
```

Comment:
Should this be gated, `MembershipState::Knock` is not.

Reply:
Well, `room_version.allow_knocking` is feature-gated so this has to be. Otherwise it won't compile.

Reply:
Good point. Should the variant also be gated as it is only part of the unstable spec or is that close enough?

Reply:
The variant has been part of the stable spec for a long time, just without any meaning attached.